CMakeLists.txt

```diff
@@ -2337,7 +2338,16 @@ endif ()
 
 if (is_multiconfig)
   # Rippled headers. Only useful for IDEs.
-  file (GLOB_RECURSE rippled_headers src/*.h src/*.hpp *.md)
+  file (GLOB_RECURSE rippled_headers RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
+    CONFIGURE_DEPENDS src/*.* *.md)
+  foreach (_target secp256k1 ed25519-donna rocksdb pbufs xrpl_core rippled)
```

Comment:
Here's an alternative approach for you consideration:
```
if (is_multiconfig)
  # Rippled headers. Only useful for IDEs.
  file (GLOB_RECURSE all_source RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
    CONFIGURE_DEPENDS src/*.* *.md)
  add_custom_target (all_sources SOURCES ${all_source})
  if (MSVC)
    set_property(
      DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
      PROPERTY VS_STARTUP_PROJECT rippled)
  endif ()
  group_sources(src)
  group_sources(docs)
  group_sources(Builds)
  group_sources(${nih_cache_path})
endif ()
```
...which just creates a dummy custom target. custom targets are not build by default so I don't expect any negative interaction. The one nice thing about this approach is that there is zero chance of marking a source file as HEADER_ONLY when you do in-fact want it to be compiled (say if the list removal is broken for some reason).